src/main/java/org/broadinstitute/dsm/model/PDF/DownloadPDF.java

```diff
@@ -0,0 +1,291 @@
+                }
+            }
+            else {
+                throw new RuntimeException("DDPInstance of participant " + ddpParticipantId + " not found");
+            }
+            response.status(200);
+        }
+        else {
+            response.status(500);
+            throw new RuntimeException("Error missing ddpParticipantId");
+        }
+        return response;
+
+    }
+
+    private byte[] generateSinglePDF(@NonNull String requestBody, String configName, UserDto user, DDPInstance ddpInstance) {
+        byte[] pdfByte = null;
+        if (COVER.equals(configName)) {
+            pdfByte = new MRCoverPDF(this).getMRCoverPDF(requestBody, ddpInstance, user);
+        }
+        else if (IRB.equals(configName)) {
+            String groupId = DDPInstance.getDDPGroupId(ddpInstance.getName());
+            pdfByte = PDFProcessor.getTemplateFromGoogleBucket(groupId + "_IRB_Letter.pdf");
+        }
+        else if (TISSUE.equals(configName) || REQUEST.equals(configName)) {
+            TissueCoverPDF tissueCoverPDF = new TissueCoverPDF(this);
+            pdfByte = tissueCoverPDF.getTissueCoverPDF(ddpInstance, user);
+        }
+        else if (configName != null) {
+            pdfByte = requestPDF(ddpInstance, ddpParticipantId, configName);
+        }
+        return pdfByte;
+    }
+
+    private byte[] getPDFBundle(DDPInstance ddpInstance, String requestBody, UserDto user) {
+        if (ddpInstance != null && StringUtils.isNotBlank(ddpParticipantId)) {
+            ByteArrayOutputStream output = new ByteArrayOutputStream();
+            try {
+                PDFMergerUtility pdfMerger = new PDFMergerUtility();
+                pdfMerger.setDestinationStream(output);
+                pdfs.sort(Comparator.comparing(org.broadinstitute.dsm.model.ddp.PDF::getOrder));
+
+                //make cover pdf first
+                if (pdfs != null && pdfs.size() > 0) {
+                    pdfs.forEach(pdf -> {
+                                byte[] pdfByte;
+                                if (pdf.getOrder() > 0) {
+                                    pdfByte = generateSinglePDF(requestBody, pdf.getConfigName(), user, ddpInstance);
+                                    pdfMerger.addSource(new ByteArrayInputStream(pdfByte));
+                                }
+                            }
+                    );
+                }
+                pdfMerger.mergeDocuments(MemoryUsageSetting.setupTempFileOnly());
+                //todo get page count and add them to the cover/request pdf
+            }
+            catch (IOException e) {
+                throw new FileProcessingException("Unable to merge documents ", e);
+            }
+            return output.toByteArray();
+        }
+        return null;
+    }
+
+    protected byte[] generatePDFFromValues(Map<String, Object> valueMap, DDPInstance ddpInstance, PDFProcessor processor) {
+        InputStream stream = null;
```

Comment:
`InputStream` has type of `AutoCloseable`, that's why you can use `try-with-resource` construction which is responsible to automatically close every resource, thus it gives you much cleaner code than this `finally` and closing resources manually.
```
try (InputStream stream = processor.generateStream(valueMap)) {
            stream.mark(0);
            return IOUtils.toByteArray(stream);
} catch (IOException e) {
            throw new RuntimeException("Couldn't get pdf for participant " + ddpParticipantId + " of ddpInstance " + ddpInstance.getName(), e);
}
```